scheduler/distributed_task_test.go

```diff
@@ -112,24 +114,34 @@ func TestDistributedWorkflow(t *testing.T) {
 			wf := dsWFMap(port1)
 			controlproxy.MAX_CONNECTION_TIMEOUT = 10 * time.Second
 			interval := time.Millisecond * 100
-			t, errs := sch.CreateTask(schedule.NewSimpleSchedule(interval), wf, true)
+			t, errs := sch.CreateTask(schedule.NewSimpleSchedule(interval), wf, true, core.OptionStopOnFailure(1), core.TaskDeadlineDuration(5*time.Millisecond))
 			So(len(errs.Errors()), ShouldEqual, 0)
 			So(t, ShouldNotBeNil)
 			c2.Stop()
-			// Give task time to fail
-			time.Sleep(time.Second)
-			tasks := sch.GetTasks()
-			var task core.Task
-			for _, v := range tasks {
-				task = v
-			}
-			So(task.State(), ShouldEqual, core.TaskDisabled)
+			// setup an event handler and wait for the 'TaskDisabledEvent' to arrive
+			respChan := make(chan bool)
+			sch.RegisterEventHandler("test", &failHandler{respChan})
+			ok := <-respChan
```

Comment:
let's chat about this offline, @IRCody; @danielscottt and I were trying to get the test to pass using a select with a timeout and seemed to see the same sort of intermittent failures that we saw when the `time.Sleep(time.Second)` statement was used to try to give the task time to fail enough times to be disabled by the scheduler. Must be some sort of race condition in how the tasks are disabled, but didn't want to dig into the scheduler to try to find the root cause (and even if I knew where to look I assumed the fix wouldn't really make sense as part of this PR and should be submitted as a separate PR focused specifically on that issue).

That being said, I'm fully supportive of trying to describe the problem we uncovered in this test as part of a separate issue (tagged/labeled as a bug?) to make sure we don't lose track of it. Like I said, let's discuss it and see if we can describe what the problem is as a new issue that can be tracked moving forward.